.github/workflows/_run_tests.yml

```diff
@@ -0,0 +1,45 @@
+# reusable workflow to run all tests
+name: run-tests
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        required: true
+        type: string
+      python-version:
+        required: true
+        type: string
+      install-script:
+        required: true
+        type: string
+jobs:
+  pre-commit:
+    runs-on: ${{ inputs.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: conda-incubator/setup-miniconda@v3
+      with:
+        miniconda-version: "latest"
+        auto-update-conda: true
+        python-version: ${{ inputs.python-version }}
+    - name: Conda info
+      shell: bash -l {0}
+      run: conda info
+    - name: Perform pip installation with all stable dependencies
```

Comment:
(nit) Perform pip installation with selected (stable/loose) dependencies

Reply:
We can create another PR for this

Reply:
Actually, the stable/loose is controlled in pip_intallation.yml, not here